wireapp/wire-ios-mono
# fix: required migration stop notification extension [WBP-5825]

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

The notification extension does not do database migration itself, but the main app. Unfortunately the extension still tried to process events even though the database migration is required and events cannot be stored.

### Solutions

Throw an error if core data migration is needed.

### Testing

#### Test Coverage (Optional)


#### How to Test

- Override `needsMigration` with true
- Run the app on your device and put it to background
- Send a push notification
- The notification should not be scheduled to the user

The new exception is caught and logged:
<img width="1903" alt="Screenshot 2023-12-08 at 14 51 57" src="https://github.com/wireapp/wire-ios/assets/7196958/386126a6-1a04-47b4-aad9-7f3b78bb8a47">


----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
